File: javascript/hamming/hamming.js
function compute(string1, string2){
  if(string1.length !== string2.length)
    throw 'DNA strands must be of equal length.';
  
  return string1.split('').reduce(function(counter, element, index){
    if(element!==string2[index])  counter++; 
    return counter;
  },0);
}

module.exports.compute = compute;

File: ruby/hamming/hamming.rb
class Hamming
  VERSION=1
  def self.compute(string1, string2)
    raise ArgumentError if string1.length != string2.length
    counter=0
    string1.length.times do |i|
      counter+=1 if string1[i]!=string2[i]
    end
    return counter
  end
end